src/data.py

```diff
@@ -38,27 +38,52 @@ def check_sample_images(directory: str) -> Tuple[List[str], int]:
     return empty_or_corrupted, num_correct
 
 
-def create_data_subsets(subsets_path: str, labels_path: str) -> None:
+def prepare_dataframe(subsets_path: str, labels_path: str) -> pd.DataFrame:
```

Comment:
imo the name of this function could be more specific e.g. prepare_merged_dataframe